testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java

```diff
@@ -397,26 +414,33 @@ public static void carryOutGDBSession(StringBuffer stringBuffer, GDBSession gdbS
                                     final String url = cp.c.split("URL ")[1];
                                     final String content = WebpageTester.getUrlContents(url);
                                     if (!cp.p.matcher(content).matches()) {
-                                        errorQueue.add("Content of URL " + url + " should have matched regexp " + cp.p.pattern() + " but it was this: " + content);
+                                        errorQueue.add("Content of URL " + url + " should have matched regexp " + cp.p.pattern() +
+                                                " but it was this: " + content);
                                     }
+                                    gotoException = null;
                                 } catch (IOException e) {
                                     gotoException = e;
                                 }
                             };
-                            int tryCount = MAX_GOTO_TRIES;
+                            final long gotoURLStart = System.currentTimeMillis();
+                            final long timeoutMs = (UsedVersion.getVersion(inContainer)
+                                    .compareTo(Version.create(23, 0, 0)) >= 0) ? LONG_GOTO_URL_TIMEOUT_MS : GOTO_URL_TIMEOUT_MS;
+                            long duration = 0;
                             do {
-                                tryCount--;
                                 esvc.submit(webRequest);
```

Comment:
```suggestion
                                gotoException = null;
                                esvc.submit(webRequest);
```

Reset `gotoException` from the `main` thread and only allow it to be `set` by the request thread. This way we can't have multiple threads running at the same time and altering the variable.

Right now we can have main spawn a request `T1` which will fail, then `main` will spawn a new request `T2` while keeping the `gotoException` set. Assuming `T2` succeeds, it will hit the breakpoint which takes more than `GOTO_URL_SLEEP_MS`. As a result `T2` is still active but hasn't reset `gotoException`, `main` doesn't wait for it and keeps spawning `T3`, `T4`, etc. which all fail and constantly set `gotoException` possibly overriding `T2`'s reset when it completes.

Reply:
You are correct. It was exactly this problem.